### PR TITLE
Add "raw" body request/response sub-tabs to Request tab.

### PR DESCRIPTION
**Pull request by philliphoff**
Adds a "raw" sub-tab to the request/response tabs within the Request tab, displaying the un-formatted, un-highlighted body content of the request/response.

screen shot 2016-06-24 at 9 23 51 am

### CODE REVIEW

**Comment by philliphoff** [comment]
Associated with #135.

File: src/request/components/RequestDetailPanelRequest.tsx [hunk 1]
```diff
@@ -79,4 +82,12 @@ export class Request extends React.Component<IRequestProps, {}> {
             </div>
         );
     }
+
+    private renderRaw(body: string) {
+        return (
+            <div className='tab-request-raw'>
```
Comment (avanderhoorn) [inline src/request/components/RequestDetailPanelRequest.tsx]: nit, quote usage.

**Comment by avanderhoorn** [comment]
Looks great, just the quote usage nit.

File: src/request/components/RequestDetailPanelRequest.tsx (hunk 1 above)
Comment (philliphoff) [inline src/request/components/RequestDetailPanelRequest.tsx]: TypeScript leans towards single quotes `'` to delimit strings and we have linting rules to that effect (at least in other repos).  Unless, you mean something entirely different.

**Comment by avanderhoorn** [comment]
I just realized, this quite the right functionality for the tab. This tab is supposed to show the "whole" raw req/res. Currently this just shows the raw body, but the whole raw req/res also includes the heads, body, etc.

**Comment by nikmd23** [comment]
Yeah, at @avanderhoorn just mentioned, this should show the raw HTTP request or response.

**Comment by avanderhoorn** [comment]
Are you happy that this is a reconstruction vs an exact copy? An exact copy would require us to essentially have all the data duplicated.

**Comment by nikmd23** [comment]
I'm happy with it being a reconstruction. Just watch things like the sort order for headers, and capitalization, we'll want to get those right.

**Comment by avanderhoorn** [comment]
Re sort order... I'm not 100% sure if we get in at the level where sort order is maintained. Additionally on the response we may not get all headers since some headers are added downstream of where we tap.

**Comment by philliphoff** [comment]
> I'm happy with it being a reconstruction. Just watch things like the sort order for headers, and capitalization, we'll want to get those right.

We won't necessarily get the actual sort order for headers, as it's just a map with no guarantees about enumeration order.  (We could theoretically capture set order within the application for responses, but that's not necessarily transmittal order.) The proxies also do not, currently capture the exact capitalization of set headers (as the spec says they are case-insensitive way).

**Comment by avanderhoorn** [comment]
Closing for the time being.

**Pull request closed without merging**